src/rcore.c

```diff
@@ -436,8 +436,7 @@ typedef struct CoreData {
 #endif
         struct {
             int exitKey;                    // Default exit key
-            char currentKeyState[MAX_KEYBOARD_KEYS];        // Registers current frame key state
-            char previousKeyState[MAX_KEYBOARD_KEYS];       // Registers previous frame key state
+            char keyState[MAX_KEYBOARD_KEYS];               // Registers frame key state
```

Comment:
@ubkp, correct me if im wrong. The issue is this change? I was unware CORE was used so oftin if this is the case. I can make some changes to the pr if you can be more specific with what the issue is.

Reply:
@n77y That too, but I was mostly looking at `FLAGS_CHECK(KEY_STATE(key)...` and `KEY_STATE(...`. Right now I can tap into anything `CORE.*` and extend it directly/easily without breaking anything and keeping fully upstream compatibility. Your PR proposes a different logic, it changes too much.

IMHO, _keypress repeated_ should be handled by the user (it's not a one size fits all), but if that's to be added, I'd much rather an approach that keeps it mostly isolated/independent to not break what is already working really well. That's why I mentioned https://github.com/raysan5/raylib/pull/3245.

Reply:
@ubkp, I have taken your requests and staged them.

Reply:
@n77y That's looking so much better. But why are you implementing a `KEY_SAFE(...`?

Personally, I don't think that making `IsKeyPressed()` require `IsKeyDown()` and `IsKeyReleased()` require `IsKeyUp()` would be a good idea, makes them dependent. IMHO, the current implementation is very resilient due to being independent of each other. Does those functions really need to change?

Reply:
@ubkp, I believe ```KEY_SAFE(key)``` to be necessary due to the exposed nature of the ```IsKey...```'s. It only seems mandatory to ensure that we aren’t reading memory we aren’t supposed to and rather simply return ```false``` when the user inputs a invalid key.

I agree, removing dependencies on the ```IsKey...```'s is sensible. I removed them.

Reply:
```KEY_SAFE(key)``` macro removed. Replaced with normal safety checks

Reply:
@n77y Regarding the key safe check, the first sentence of this comment https://github.com/raysan5/raylib/issues/3246#issuecomment-1682256790 comes to mind. So I assumed it was up to the user to validate that. I'm not against it tho, I think that's good.

Just tested your PR on `PLATFORM_DESKTOP` (Linux Mint 21.1 64-bit) and `PLATFORM_WEB` (Chromium 115.0.5790.170 64-bit and Firefox 115.1.0esr 64-bit, both on Linux), and it's working great. Very good job.

---

Tested with:
```
#include "raylib.h"

int main(void) {
    InitWindow(800, 450, "test");
    SetTargetFPS(60);
    while (!WindowShouldClose()) {

        if (IsKeyRepeated(KEY_ONE)) TraceLog(LOG_INFO, "1 repeated");

        BeginDrawing();
        ClearBackground(RAYWHITE);
        EndDrawing();
    }
    CloseWindow();
    return 0;
}
```